lms/static/js/learner_dashboard/EnterpriseLearnerPortalModal.jsx

```diff
@@ -0,0 +1,126 @@
+    if (!hasViewedModal) {
+      this.openModal();
+      document.addEventListener('mousedown', this.handleClick, false);
+      window.sessionStorage.setItem(storageKey, true);
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.isModalOpen !== prevState.isModalOpen) {
+      if (this.state.isModalOpen) {
+        // add a class here to prevent scrolling on anything that is not the modal
+        document.body.classList.add('modal-open');
+      } else {
+        // remove the class to allow the dashboard content to scroll
+        document.body.classList.remove('modal-open');
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    // remove the class to allow the dashboard content to scroll
+    document.body.classList.remove('modal-open');
+    document.removeEventListener('mousedown', this.handleClick, false);
+  }
+
+  handleClick(e) {
+    if (this.modalRef && this.modalRef.contains(e.target)) {
+      // click is inside modal, don't close it
+      return;
+    }
+
+    this.closeModal();
+  }
+
+  closeModal() {
+    this.setState({
+      isModalOpen: false,
+    });
+  }
+
+  openModal() {
+    this.setState({
+      isModalOpen: true,
+    });
+  }
+
+  getLearnerPortalUrl() {
+    const baseUrlWithSlug = `${this.props.enterpriseLearnerPortalBaseUrl}/${this.props.enterpriseCustomerSlug}`;
+    return `${baseUrlWithSlug}?utm_source=lms_dashboard_modal`;
+  }
+
+  render() {
+    if (!this.state.isModalOpen) {
+      return null;
+    }
+
+    return (
+      <div
+        role="dialog"
+        className="modal-wrapper d-flex align-items-center justify-content-center"
+      >
+        <FocusLock>
+          <div
+            className="modal-content p-4 bg-white"
+            ref={(node) => { this.modalRef = node; }}
+          >
+            <div className="mb-3 font-weight-bold">
+              {StringUtils.interpolate(
+                gettext('You have access to the {enterpriseName} dashboard'),
+                {
+                  enterpriseName: this.props.enterpriseCustomerName,
+                }
+              )}
+            </div>
+            <p>
+              {StringUtils.interpolate(
+                gettext('To access the courses available to you through {enterpriseName}, visit the {enterpriseName} dashboard.'),
+                {
+                  enterpriseName: this.props.enterpriseCustomerName,
+                }
+              )}
+            </p>
+            <div className="mt-4 d-flex align-content-center justify-content-end">
+              <button
+                className="btn-link mr-3"
+                onClick={() => this.closeModal()}
+              >
+                {gettext('Cancel')}
+              </button>
+              <AutoFocusInside>
+                <a
+                  href={this.getLearnerPortalUrl()}
+                  className="btn btn-primary"
+                >
+                  {gettext('Go to dashboard')}
```

Comment:
Nit: Maybe "Go to {enterpriseName} dashboard"?  I know it's gonna be longer, but maybe it would help avoid possible confusion around the current page's URL being `/dashboard`?  Please ignore if we already made a UX/copy decision about this.

Reply:
As you called out, the button text would be quite long in some cases due to lengthy enterprise names (see screenshot as an example). The other messaging in the modal body mentions the enterprise name in association with "dashboard" which should reduce this potential confusion.

![image](https://user-images.githubusercontent.com/2828721/119861019-ecd2f000-bee4-11eb-8264-5a02e72c6548.png)

Reply:
😆 In the words of Brandon, ignore me.